modules/scanning-delegate-role/main.tf

```diff
@@ -531,3 +579,15 @@ resource "aws_iam_role_policy_attachment" "workers_dspm_attachment" {
   policy_arn = aws_iam_policy.scanning_worker_dspm_policy[0].arn
   role       = aws_iam_role.role.name
 }
+
+resource "aws_iam_role_policy_attachment" "workers_bucket_attachment" {
+  count      = var.sensitive_data_scanning_enabled ? 1 : 0
+  policy_arn = aws_iam_policy.scanning_worker_bucket_policy[0].arn
+  role       = aws_iam_role.role.name
+}
+
+resource "aws_s3_bucket" "bucket" {
```

Comment:
this will trigger a number of CSPM rules (including our own), we should make sure that basic encryption is turned on and there's a public access block attached with all settings to true

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/s3_bucket_public_access_block
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/s3_bucket_server_side_encryption_configuration